src/ot_orchestration/utils/utils.py

```diff
@@ -37,47 +37,87 @@ def clean_name(name: str) -> str:
     return re.sub(r"[^a-z0-9-]", "-", name.lower())
 
 
-def create_name(prefix: str, suffix: str) -> str:
-    """Create a clean name meeting google cloud naming conventions."""
-    return re.sub(r"[^a-z0-9-]", "-", f"{prefix}-{suffix}".lower())
+def create_name(step_name: str) -> str:
+    """Create a google resource name for a given step name."""
+    return f"up-{clean_name(step_name)}-{{{{ run_id | strhash }}}}"
+
+
+def create_cluster_name(task_group_name: str) -> str:
+    """Create a cluster name for a given task group name.
+
+    The name will include our prefix `up-` and the task group name, so for the
+    gentropy stage of run `3beef`, the cluster name will be:
+
+    `up-gentropy-3beef`
+    """
+    return f"up-{clean_name(task_group_name)}-{{{{ run_id | strhash }}}}"
 
 
-def read_yaml_config(config_path: Path | str) -> Any:
-    """Parse a YAMl config file and do all necessary checks.
+def read_yaml_config(
+    config_path: Path | str,
+    sentinels: dict[str, str] | None = None,
+) -> Any:
+    """Parse a YAML config file replacing sentinels.
 
     Args:
         config_path (Path | str): Path to the YAML config file.
+        sentinels (dict[str, str] | None): Sentinels to replace in the config file.
+
+    Sentinels in yaml files are in the form:
+
+    `{variable_name}`
 
     Returns:
         Any: Parsed YAML config file.
     """
     config_path = config_path if isinstance(config_path, Path) else Path(config_path)
     assert config_path.exists(), f"YAML config path {config_path} does not exists"
-    with open(config_path) as config_file:
-        return yaml.safe_load(config_file)
+
+    raw_config = config_path.read_text()
+    if sentinels:
+        for sentinel, replacement in sentinels.items():
+            raw_config = raw_config.replace(f"{{{sentinel}}}", replacement)
```

Comment:
why not use the str.format directly? it will be less ugly

Reply:
It's not that easy, because there are other strings that look like sentinels in the file, but are not meant to be replaced. [The way `str.format()` works](https://docs.python.org/3/library/stdtypes.html#strw.format) is, it looks for all the strings in braces in the sentinels dict.

In our case, it will fail because it won't find the key for:

```
+step.colocalisation_method_params: '{priorc1: 1e-4, priorc2: 1e-4, priorc12: 1e-5}'
```

Which is not meant to be replaced anyway.

Reply:
Same as above, I agree.